ra/registration-authority.go

```diff
@@ -121,23 +121,38 @@ func validateEmail(ctx context.Context, address string, resolver bdns.DNSResolve
 	domain := strings.ToLower(splitEmail[len(splitEmail)-1])
 	var resultMX []string
 	var resultA []net.IP
-	resultMX, err = resolver.LookupMX(ctx, domain)
-	if err == nil && len(resultMX) == 0 {
-		resultA, err = resolver.LookupHost(ctx, domain)
-		if err == nil && len(resultA) == 0 {
-			return &probs.ProblemDetails{
-				Type:   probs.InvalidEmailProblem,
-				Detail: emptyDNSResponseDetail,
-			}
-		}
+	var errMX, errA error
```

Comment:
Hmm, I  kinda like it how it is right now because the code has less indents.

It also doesn't behave differently when you have MX succeed but A error.